cpp/oneapi/dal/algo/decision_forest/common.cpp

```diff
@@ -0,0 +1,182 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/algo/decision_forest/common.hpp"
+#include "oneapi/dal/algo/decision_forest/detail/model_impl.hpp"
+
+namespace oneapi::dal::decision_forest {
+
+class detail::descriptor_impl : public base {
+public:
+    double observations_per_tree_fraction      = 1.0;
+    double impurity_threshold                  = 0.0;
+    double min_weight_fraction_in_leaf_node    = 0.0;
+    double min_impurity_decrease_in_split_node = 0.0;
+
+    std::int64_t class_count                    = 1;
```

Comment:
1 class as default? 😄  Will random forest work?

Reply:
good catch ), thanks.

Reply:
yes, it should be 2

Reply:
fixed